Fix exported types conflicting with agent protocol

pkukielka opened the pull request:
## Test plan

Compile JetBrains repo with CODY_DIR set to checked out code from this branch.
There should be no errors.

pkukielka commented on agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/AuthStatus.kt:
```diff
@@ -1,11 +1,41 @@
+import com.google.gson.JsonElement;
+import java.lang.reflect.Type;
+
+sealed class AuthStatus {
```
Deserialisation code is incorrect, we need to fix this before merging.

pkukielka replied:
I patched it manually. It is not perfect but should be enough for a temp solution.

dominiccooney replied:
Is it incorrect because it is just out of date, or is it more fundamental?

If there's a bigger issue here someone fixing CODY-3809 needs to work on, could you drop a sketch of the issue in the comment, maybe make it TODO(CODY-3809): ...

pkukielka replied:
It is a bigger issue, I already discussed it with @olafurpg and @sqs on slack.
I will add comment in the linear.

dominiccooney commented on agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/AuthStatus.kt:
```diff
@@ -1,21 +1,59 @@
+/*
```
You could review https://github.com/sourcegraph/cody/pull/5773 so people are aware when they are breaking JetBrains (...eventual goal to make that land blocking.)

dominiccooney commented on agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/AuthStatus.kt:
```diff
@@ -1,21 +1,59 @@
+import com.google.gson.JsonElement;
+import java.lang.reflect.Type;
 
+// TODO: This was edited manually due to definiencies in the codegen.
```
Spelling, deficiencies

pkukielka replied:
Fixed.

dominiccooney replied:
Still not right: definiencies -> deficiencies

dominiccooney reviewed (approved):
Approving to unblock because this is part of a multi-sided patch to unblock JetBrains release.

However we are compounding tech debt here, which is not great. I'm concerned that there is no test plan, so how will we ensure this doesn't break when CODY-3089 is fixed? Could you:

- Add a test plan
- Mention TODO(CODY-3089) in your comment, and add some details about what is hand-edited here
- Drop a note in CODY-3089 pointing to this PR and noting that we need to run the test plan after switching back to the generated bindings.

pkukielka commented:
> Approving to unblock because this is part of a multi-sided patch to unblock JetBrains release.
> 
> However we are compounding tech debt here, which is not great. I'm concerned that there is no test plan, so how will we ensure this doesn't break when [CODY-3089](https://linear.app/sourcegraph/issue/CODY-3089) is fixed? Could you:
> 
> * Add a test plan
> * Mention TODO([CODY-3089](https://linear.app/sourcegraph/issue/CODY-3089)) in your comment, and add some details about what is hand-edited here
> * Drop a note in [CODY-3089](https://linear.app/sourcegraph/issue/CODY-3089) pointing to this PR and noting that we need to run the test plan after switching back to the generated bindings.

1. I'm not sure what testplan it should be.
TBH this would and should be caught by automatic binding regeneration.
We ran into this issue because it was disabled. 
2. I already mentioned CODY-3089 in my comment before but I made it more explicit now.
3. I added a note in linear (and changed it title to be more truthful) 👍

dominiccooney commented:
Wow, JetBrains FYI did its thing: https://github.com/sourcegraph/cody/actions/runs/11175844736/job/31067972094?pr=5794

pkukielka commented:
> Wow, JetBrains FYI did its thing: https://github.com/sourcegraph/cody/actions/runs/11175844736/job/31067972094?pr=5794

This requires fix on the jetbrains side, I will merge this PR and soon raise a fix with a cody commit bump.
